### ISSUE TITLE
LibreCat::Model::User::find too complex

### ISSUE THREAD

**Issue by nicolasfranck** [LibreCat/LibreCat#908]
The methode `find` in `LibreCat::Model::User` executes too many possible queries:

* two possible queries against ElasticSearch (search on id, and if that fails, search on login).
* a `get` against the datastore itself
* a `find_by_username`, of which the number of search requests is determined by the configuration.

This method is only used by `LibreCat::App::Helper::Helpers::get_person`, and only `LibreCat::App::Catalogue::Route::file` uses any other argument than the user_id: email.

What to do with this?

**Comment by nicolasfranck** [comment]
This method is used by `LibreCat::App::Helper::Helpers::get_person`. I've tried to replace all instances of `get_person` by `main_user->get` where I was sure that the real user identifier was given: https://github.com/LibreCat/LibreCat/pull/910

**Comment by nicolasfranck** [comment]
crap. Just realised that not all users are stored in the main user table. Sigh.

**Comment by vpeil** [comment]
> crap. Just realised that not all users are stored in the main user table. Sigh.

Do you mean the case we discussed in #748 ?

**Comment by nicolasfranck** [comment]
@vpeil yes. That ticket also explains that we have users that we have no control over, as we cannot change them.
